extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/RestClientBuilderHelper.java

```diff
@@ -77,7 +81,15 @@ public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpCli
                     httpClientBuilder.setSSLStrategy(NoopIOSessionStrategy.INSTANCE);
                 }
 
-                return httpClientBuilder;
+                // Apply configuration from RestClientBuilder.HttpClientConfigCallback implementations annotated with ElasticsearchClientConfig
+                HttpAsyncClientBuilder result = httpClientBuilder;
+                for (InstanceHandle<RestClientBuilder.HttpClientConfigCallback> handle : Arc.container()
+                        .select(RestClientBuilder.HttpClientConfigCallback.class, new ElasticsearchClientConfig.Literal())
+                        .handles()) {
```

Comment:
Can you split this line in two to improve readibility ? I have a hard time understanding what it is about at first read.
```suggestion
                List<InstanceHandle<RestClientBuilder.HttpClientConfigCallback>> handles =  Arc.container()
                        .select(RestClientBuilder.HttpClientConfigCallback.class, new ElasticsearchClientConfig.Literal())
                        .handles();
                for (InstanceHandle<RestClientBuilder.HttpClientConfigCallback> handle : handles) {
```